metricbeat/module/windows/perfmon/perfmon.go

```diff
@@ -0,0 +1,77 @@
+// +build windows
+
+package perfmon
+
+import (
+	"errors"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/metricbeat/mb"
+)
+
+type CounterConfig struct {
+	Name  string               `config:"group" validate:"required"`
+	Group []CounterConfigGroup `config:"collectors" validate:"required"`
+}
+
+type CounterConfigGroup struct {
+	Alias string `config:"alias" validate:"required"`
+	Query string `config:"query" validate:"required"`
+}
+
+// init registers the MetricSet with the central registry.
+// The New method will be called after the setup of the module and before starting to fetch data
+func init() {
+	if err := mb.Registry.AddMetricSet("windows", "perfmon", New); err != nil {
+		panic(err)
+	}
+}
+
+// MetricSet type defines all fields of the MetricSet
+// As a minimum it must inherit the mb.BaseMetricSet fields, but can be extended with
+// additional entries. These variables can be used to persist data or configuration between
+// multiple fetch calls.
+type MetricSet struct {
+	mb.BaseMetricSet
+	counters []CounterConfig
+	handle   *Handle
+}
+
+// New create a new instance of the MetricSet
+// Part of new is also setting up the configuration by processing additional
+// configuration entries if needed.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+
```

Comment:
A log message should be added to make it clear it is beta: https://github.com/elastic/beats/blob/master/metricbeat/module/jolokia/jmx/jmx.go#L49 We do this for all new modules / metricsets to test them first.